### PR TITLE
[Runtime] Fix _swift_validatePrespecializedMetadata when type lookup fails.

### PR DESCRIPTION
**Pull request by mikeash**
When we fail to look up a type by name, we print an error, then try to compare anyway, which crashes. Skip the comparison when that happens.

While we're in there, modify _swift_validatePrespecializedMetadata to be more useful for debugging, by removing the parameters and having it print the results directly.

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please test

File: stdlib/public/runtime/LibPrespecialized.cpp [hunk 1]
```diff
@@ -186,12 +182,16 @@ void _swift_validatePrespecializedMetadata(unsigned *outValidated,
               "Prespecializations library validation: unable to build metadata "
               "for mangled name '%s'\n",
               mangledName);
-      if (outFailed)
-        (*outFailed)++;
+      failed++;
+      continue;
```
Comment (al45tair) [inline stdlib/public/runtime/LibPrespecialized.cpp:186]: :-) This line is presumably the fix.

File: stdlib/public/runtime/LibPrespecialized.cpp [hunk 2]
```diff
@@ -186,12 +182,16 @@ void _swift_validatePrespecializedMetadata(unsigned *outValidated,
...
     if (!compareGenericMetadata(result.getType().getMetadata(), element.value))
-      if (outFailed)
-        (*outFailed)++;
+      failed++;
   }
+
+  fprintf(stderr,
+          "Prespecializations library validation: validated %u entries, %u "
+          "failures.\n",
+          validated, failed);
```
Comment (al45tair) [inline stdlib/public/runtime/LibPrespecialized.cpp:196]: Did you mean to leave this debug output in?

**Review by al45tair — APPROVED** [review]

File: stdlib/public/runtime/LibPrespecialized.cpp (hunk 2 above)
Comment (mikeash) [reply stdlib/public/runtime/LibPrespecialized.cpp:196]: Yes, this is meant to be called manually from a debugger now.

File: stdlib/public/runtime/LibPrespecialized.cpp (hunk 1 above)
Comment (mikeash) [reply stdlib/public/runtime/LibPrespecialized.cpp:186]: Yeah, it's a bit buried.

**Comment by mikeash** [comment]
I forgot to change the prototype in the header, which caused the symbol checks to fail since the renamed function was no longer exported. Nice test coverage on that! Fixed it now.

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test

**Pull request merged**